optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/statistic/SingleStatistic.java

```diff
@@ -150,6 +150,9 @@ public void readCsvStatisticFile() {
             }
             Map<String, String> stringDuplicationRemovalMap = new HashMap<String, String>(1024);
             for (line = reader.readLine(); line != null && !line.isEmpty(); line = reader.readLine()) {
+                if (line.equals("Failed")) { // Reading statistics of a failed benchmark
+                    break;
```

Comment:
I would expect it to do either of these 2 things when it finds "Failed":
- set a flag to mark this single statistic as failed (I presume that it already knows that through the benchmark xml loaded earlier)
- Check if that flag was already set (see above) and fail-fast if it didn't.